### ISSUE TITLE
Modal dialog accessibility

### ISSUE THREAD

**Issue by PVince81** [nextcloud/nextcloud-vue#2501]
* [ ] The div needs to have `role="dialog"`: https://github.com/nextcloud/nextcloud-vue/issues/1590
  * [ ] Name of dialog needs to be communicated, e.g. via `aria-labelledby` linked to the dialog title, which ideally is an h2 inside the dialog
  * [ ] Keep focus within the dialog, prevent tabbing through elements in background: https://github.com/nextcloud/nextcloud-vue/issues/1591
  * [ ] Navigation elements should be visible at all times, not fade out
  * [ ] Prev/next arrows are not focusable at all, and should be buttons
  * [ ] Close button does not seem to work on enter, focus moves to other element
  * [ ] On closing, focus has to move back to element used to open the dialog – show clear visible focus on that again
  * [ ] Image viewer zoom action should ideally have the image inside a button, with `aria-label` description
  * [ ] Image viewer needs properly set `alt` tag
  * [ ] Settings modal should always have close button on top right inside the modal, which should not hide

**Comment by vanpertsch** [comment]
https://github.com/nextcloud/nextcloud-vue/pull/2654

**Comment by PVince81** [comment]
I've transferred the "image viewer" tickets to the viewer repo

originally that checkbox list was a dump of my findings where I was looking at various modals and their contents

**Comment by PVince81** [comment]
remaining task, was a result of an accessibility review and an answer to the question whether we should automatically focus inside a dialog that was just opened: https://github.com/nextcloud/nextcloud-vue/issues/3088

**Comment by JuliaKirschenheuter** [comment]
all closed
